client/core/trade_simnet_test.go

```diff
@@ -181,10 +181,11 @@ func TestTrading(t *testing.T) {
 
 	// run subtests
 	tests := map[string]func(*testing.T){
-		"success":         testTradeSuccess,
-		"no maker swap":   testNoMakerSwap,
-		"no taker swap":   testNoTakerSwap,
-		"no maker redeem": testNoMakerRedeem,
+		"success":                  testTradeSuccess,
+		"no maker swap":            testNoMakerSwap,
+		"no taker swap":            testNoTakerSwap,
+		"no maker redeem":          testNoMakerRedeem,
+		"maker ghost after redeem": testMakerGhostingAfterTakerRedeem,
 	}
 
 	for test, testFn := range tests {
```

Comment:
This seems to be the problem as ranging over the map is a random order.  I suggest the following:

```go
	// run subtests
	tests := []struct {
		name string
		fun  func(*testing.T)
	}{
		{"success", testTradeSuccess},
		{"no maker swap", testNoMakerSwap},
		{"no taker swap", testNoTakerSwap},
		{"no maker redeem", testNoMakerRedeem},
		{"maker ghost after redeem", testMakerGhostingAfterTakerRedeem},
	}

	for _, test := range tests {
		fmt.Println() // empty line to separate test logs for better readability
		if !t.Run(test.name, test.fun) {
			break
		}
	}
```

Reply:
Oh right! Those btc errors are a result of not properly parsing the tx hex into a MsgTx. Will fix and push with this change

Reply:
Fixed now.